baremetal/metal3data_manager.go

```diff
@@ -992,15 +992,33 @@ func renderNetworkLinks(networkLinks infrav1.NetworkDataLink,
 		if err != nil {
 			return nil, err
 		}
-		data = append(data, map[string]interface{}{
+		entry := map[string]interface{}{
 			"type":                  "bond",
 			"id":                    link.Id,
 			"mtu":                   link.MTU,
 			"ethernet_mac_address":  macAddress,
 			"bond_mode":             link.BondMode,
 			"bond_xmit_hash_policy": link.BondXmitHashPolicy,
 			"bond_links":            link.BondLinks,
-		})
+		}
+
+		illegalParameters := []string{}
+		for opt, value := range link.Parameters {
```

Comment:
I think this would be ineffective if I would supply
"bond_mode" or "bond_xmit_hash_policy" as parameter names/keys as in this logic the "opt" is expected to **not** contain the "bond_" prefix but that expectation is not enforced anywhere AFAIK. So I would be able to overwrite the structured data still.

Do I see it correctly or did I miss something here?

Reply:
Didn't get. How you will be able to overwrite existing/structured fields? Despite presence of "bond_" prefix inside option name into parameters map key this code will add (one more if already exists) prefix "bond_" and will check is it collide with already produced fields. 

If option will have more than one "bond_" prefix, option will be ignored by cloud-init.

Reply:
Oh okay, I missed that it saves the modified name e.g it would be "bond_bond_x" propagated. Then all is fine, thanks for explaining.